docs/specs/short-read-alignment-vector.md

```diff
@@ -46,17 +46,18 @@ reads, intended for use with mosquito specimens.
 
 * Sequence reads - For each lanelet, a pair of files containing
   sequence reads, with one file containing forward reads and one file
-  containing reverse reads, in FASTQ format.
+  containing reverse reads, in gzipped FASTQ format.
```

Comment:
xref #6 

How much do we want the output of mapping to be consistent across releases?  Will we reprocess any existing samples through the new pipeline?

It's not part of the new or old specs, but the vrpipe pipelines would always filter bams before they converted them to fastq for mapping such that:

- reads that failed QC checks were removed
- non-primary alignments were removed
- chimeric (aka supplementary) alignments were removed

I assume that any fastq files released by Core would have failed reads already removed. But keeping in reads that have chimeric portions in the fastq might change the output of the new pipeline compared to previous releases.

However, if we add a step to filter out chimeric reads after mapping, then perhaps it won't be an issue.

Reply:
Hi @tnguyensanger,

IIUC, a non-primary (a.k.a. secondary) alignment in a BAM file occurs when one read gets aligned to multiple locations in the genome. If you are trying to reverse-engineer a FASTQ file from a BAM file then you would obviously want to exclude secondary alignments, otherwise you would end up with the same read output multiple times.

I understand chimeric (aka supplementary) alignments less well, but I believe this is a situation where a single read gets split up and different pieces are aligned in different genome locations. I.e., it is a result of the alignment process. So again, if you're trying to reverse-engineer a FASTQ from a BAM then you don't want these pieces coming out as if they were separate reads.

Niether of these we need to worry about if we are starting from FASTQ. I.e., starting from the original FASTQ should be equivalent to starting from a BAM and then reverse engineering a FASTQ via these options.

Regarding reads that fail QC checks, I guess you mean reads with bit 0x200 set in the FLAG field of the BAM? If so, I don't know how that flag ends up being set. If it ever is set then it has to be based on information coming from the original FASTQ in the read descriptor, i.e., based on information coming from the sequencing machine. I would bet that this flag is not being set by the aligner, and so we don't have to worry about this either. A quick `samtools flagstat` check on one of the run-level BAMs on ENA would confirm.

Hth.

Reply:
> I understand chimeric (aka supplementary) alignments less well, but I believe this is a situation where a single read gets split up and different pieces are aligned in different genome locations. I.e., it is a result of the alignment process. So again, if you're trying to reverse-engineer a FASTQ from a BAM then you don't want these pieces coming out as if they were separate reads.
>
> Niether of these we need to worry about if we are starting from FASTQ. I.e., starting from the original FASTQ should be equivalent to starting from a BAM and then reverse engineering a FASTQ via these options.

I'm assuming that when we feed fastqs reverse engineered from chimeric-alignment-free bams into bwa, the resulting bams will be mostly chimeric-alignment-free.  However, if we feed the original fastqs into bwa, the resulting bams may have more chimeric alignments.  I'm not sure how UnifiedGenotyper treats chimeric alignments.  If it ignores them, then we don't have to do anything.  But if it uses them during calling, then perhaps it would be a good step to remove the chimeric alignments before calling.  

That said, the vrpipe pipelines never did any additional filtering with the realigned bams before feeding them into UnifiedGenotyper, so perhaps this is moot.

Reply:
> I'm assuming that when we feed fastqs reverse engineered from chimeric-alignment-free bams into bwa, the resulting bams will be mostly chimeric-alignment-free. 

I think there may be a small confusion here. IIUC when a read is aligned as a chimera, one of the aligned segments is considered representative and the others are flagged as supplementary. If you are trying to recreate a fastq from a BAM, you don't want to include the supplementary alignments, because you would end up outputting the same read multiple times. In other words, excluding alignments with the "supplementary" flag is just a way of avoiding the same read being duplicated, in the same way that you want to avoid outputting reads from alignments flagged as "secondary". The read that was aligned as a chimera will still end up in the fastq, because the representative alignment will still be included. 

So I think that the process of extracting a fastq from a bam as was implemented in vr-pipe should reconstruct the fastq with *exactly* the same set of reads that were in the original fastq. I.e., there should be no difference whether you start the pipeline from bams and extract fastq, or whether you start from original fastq.

Reply:
@alimanfoo IIUC when a read is aligned as a chimera, one of the aligned segments is considered representative and the others are flagged as supplementary. 

Ah!  It was my understanding that all alignments from a chimeric read were marked as chimeric.  However, if all but one alignment are marked as chimeric, then I agree that using the original fastq as opposed to reverse engineered fastq would yield similar results in the vector pipeline.  From rereading the samtools spec, this does seem to be the case.

Reply:
Yeah, I think so. From the [SAM spec](https://samtools.github.io/hts-specs/SAMv1.pdf): "Typically, one of the linear alignments in a chimeric alignment is considered the “representative” alignment, and the others are called “supplementary” and are distinguished by the supplementary alignment flag."